### PR TITLE
Fixes #1704 - Exemplifies fix for Safari errors using async / await.

### LINKED ISSUE #1704: Get funcional tests to run in Safari and Edge locally.
So, since we are going to test these two browsers in Browserstack, I thought it would be good to be able to also test them locally.

Now, it wouldn't be really nice to have 3+ browsers run every time we test something. So, to address that issue, I thought about adding an argument where we list all browsers we want to test with when testing. Something like `browsers=firefox, chrome`.

For that we just need to get tests to run in Safari and Edge, the status right now for Safari is:

**Saf 12603 : [✓✓✓×~~✓✓✓✓××××××××✓×✓✓✓×××✓×✓××✓××××××××] 79/79, 37 fail, 3 skip**

Will add Edge here as soon as I test it, but from what I remember in testing with Browserstack, there are some errors, not as much as Safari though.

I'm ignoring IE for now because I suppose not many people from our user base are still using it. But I will confirm that info in Google Analytics.

### PR DESCRIPTION
**Pull request by brizental**
So, this was kind of a fail.

I thought using `async / await` would fix intermittent errors, but the test for `Posting a comment` still gives an error every now and then. I thought of two possibilities:

1. `await` is not working. But from what I've tested it is, maybe I'm testing wrong, idk.
2. Leadfoot functions just fail for no reason sometimes.

Not very optimistic, but yeah, lol.

Also the `Add a screenshot to a comment` test is not working for Safari. It gives back an `ElementNotVisible` error. 

Now, that I think might be a Leadfoot bug, because Leadfoot claims to be a library that fixes cross-browser inconsistencies for the SeleniumWebDriver. And this test works for Firefox and Chrome, so it should work for Safari I guess.

What makes me unsure about that, though, is the fact that in my previous PR for this issue (#1706) this test worked. 

Anyways, thoughts @miketaylr ?

### CODE REVIEW

File: tests/functional/lib/helpers.js [hunk 1]
```diff
@@ -4,135 +4,49 @@
...
-            return null;
-          }
-          elem = elem[0];
-          return elem.offsetWidth > 0 && elem.offsetHeight > 0 ? true : null;
-        },
-        [selector],
-        10000
-      );
-    }
+  async function login(context) {
```
Comment (miketaylr) [inline tests/functional/lib/helpers.js]: It seems like this login method doesn't actually do same stuff that the original one does, right?

(fill out forms, look for occasional verification page...)

Comment (brizental) [inline tests/functional/lib/helpers.js]: Ah, yes! 

I removed that just to make the transition faster for me, since this function was not the one that I actually wanted to test. 

But now I'm thinking, since we have the mock login in place is that really needed?

Comment (miketaylr) [inline tests/functional/lib/helpers.js]: Hmm..... probably not needed/used right now, but I wonder if it might be useful if we're trying to test w/o mock data? Is that even useful I wonder...

Comment (brizental) [inline tests/functional/lib/helpers.js]: I can't think of a situation where that would be needed.  But it might happen. 

Do you think I should bring it back?

**Comment by brizental** [comment]
Closing this because it doesn't work.
